opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/AbstractDaoRestService.java

```diff
@@ -275,21 +297,115 @@ public Response getProperties(@QueryParam("q") final String query) {
         }
     }
 
+    private static class HibernateListCallback<T> implements HibernateCallback<List<T>> {
+        private final SearchProperty property;
+        private final String query;
+
+        public HibernateListCallback(final SearchProperty property, final String query) {
+            this.property = property;
+            this.query = query;
+        }
+
+        @Override
+        public List<T> doInHibernate(Session session) throws HibernateException, SQLException {
+            final Query hql;
+            // TODO: Add limit?
+            // TODO: Sort by count?
+
+            // If there is a query string...
+            if (query != null && query.length() > 0) {
+                hql = session.createQuery(
+                    String.format(
+                        "select distinct %s from %s where lower(%s) like :query order by %s",
```

Comment:
The results appear sorted in the REST results (JSON and XML) so the lack of ordering must be in the JS client or higher.